ming/tests/test_mim.py

```diff
@@ -721,51 +516,95 @@ def test_distinct_sublist(self):
+                                                                                   {'i': 2},
+                                                                                   return_document=True))
+
+    def test_find_one_and_replace_fails_with_set(self):
+        self.bind.db.foo.insert_one({'_id': 1, 'i': 1})
+        with self.assertRaises(ValueError):
+            self.bind.db.foo.find_one_and_replace({'i': 1},
+                                                  {'$set': {'i': 2}},
+                                                  return_document=True)
+
+    def test_find_one_and_update_returns_new_value_on_new_filter_id(self):
+        self.bind.db.foo.insert_one({'i': 1})
+        self.assertEqual({'i': 2}, self.bind.db.foo.find_one_and_update({'i': 1},
+                                                                        {'$set': {'i': 2}},
+                                                                        projection={'_id': False, 'i': True},
+                                                                        return_document=True))
+
+    def test_find_one_and_update_returns_new_value_on_new_upsert(self):
+        self.assertEqual({'_id': 1, 'i': 2}, self.bind.db.foo.find_one_and_update({'i': 1},
+                                                                                  {'$set': {'_id': 1, 'i': 2}},
+                                                                                  return_document=True,
+                                                                                  upsert=True))
+        
+    def test_find_one_and_update_fails_with_id(self):
+        self.bind.db.foo.insert_one({'_id': 1, 'i': 1})
+        with self.assertRaises(ValueError):
+            self.bind.db.foo.find_one_and_update({'i': 1},
+                                                 {'_id': 2, 'i': 2},
+                                                 return_document=True)
+
+    def test_find_one_and_replace_returns_new_value_on_new_upsert(self):
+        doc = self.bind.db.foo.find_one_and_replace({'i': 1},
+                                                    {'i': 2},
+                                                    return_document=True,
+                                                    upsert=True)
+        self.assertIsInstance(doc.pop("_id"), bson.ObjectId)
+        self.assertEqual({'i': 2}, doc)
 
-    def test_find_and_modify_returns_new_value_on_new_upsert(self):
-        self.assertEqual({'_id': 1, 'i': 2}, self.bind.db.foo.find_and_modify({'i': 1},
-                                                                              {'$set': {'_id': 1,
-                                                                                        'i': 2}},
-                                                                              new=True,
-                                                                              upsert=True))
+    def test_find_one_and_delete(self):
```

Comment:
I think this is a duplicated test name `test_find_one_and_delete` (there's one above too).  I was looking at the codecov report and had a real hard time finding just the "missed lines" but this was one of them.  They are "missed" because pytest only runs the first test with this name.